tools/ctl/ctl.c

```diff
@@ -27,23 +27,50 @@ static void usage(char *name)
 	fprintf(stdout, " numid=22,name=\\\"EQIIR1.0 EQIIR\\\"\"\n");
 	fprintf(stdout, " -n control id e.g. 22\n");
 	fprintf(stdout, " -s set data using ASCII CSV input file\n");
+	fprintf(stdout, " -b set/get control in binary mode(e.g. for set, use binary input file, for get, dump out in hex format)\n");
+	fprintf(stdout, " -r no abi header for the input file, or not dumping abi header for get.\n");
 }
 
-static int read_setup(unsigned int *data, char setup[], size_t smax)
+static int read_setup(unsigned int *data, char file_name[], size_t smax,
+		      int binary, int raw_data)
 {
 	FILE *fh;
 	unsigned int x;
 	int n = 0;
 	int n_max = smax / sizeof(unsigned int);
 	int separator;
+	char *mode = binary ? "rb" : "r";
+	struct sof_abi_hdr *hdr;
+	int abi_size = 0;
 
+	fprintf(stdout, "read_setup(), smax:%d\n", smax);
 	/* open input file */
-	fh = fopen(setup, "r");
+	fh = fopen(file_name, mode);
 	if (!fh) {
 		fprintf(stderr, "error: %s\n", strerror(errno));
 		return -errno;
 	}
 
+	if (binary) {
+		/* create abi header*/
+		if (raw_data) {
+			hdr = data;
```

Comment:
Can we make this ABI creation a separate function, this way we can also use it for ASCII CSV data too.

Reply:
Sure, let me change it.